### PR TITLE
Make require conservative and scope protect it	

### PR DESCRIPTION
**Pull request by kelunik**

### CODE REVIEW

File: lib/Threading/Internal/Thread.php [hunk 1]
```diff
@@ -53,44 +53,44 @@ public function run() {
...
-    
+
         if ($autoloadPath === null) {
             throw new \Error('Could not locate autoload.php');
         }
-    
-        require $autoloadPath;
-        
+
+        requireFile($autoloadPath);
```
Comment (bwoebi) [inline lib/Threading/Internal/Thread.php]: `(static function() use ($autoloadPath) { require $autoloadPath; })();` will just work as well and not need an extra function.

Comment (trowski) [inline lib/Threading/Internal/Thread.php]: Using a composer-included function won't work here as it has not been defined in the thread at this point. Wrap in an immediately-invoked closure as @bwoebi suggested.

File: lib/Threading/Internal/Thread.php [hunk 2]
```diff
@@ -53,44 +53,44 @@ public function run() {
          * thread-safe objects.
          */
         $paths = [
-            \dirname(__DIR__, 5) . \DIRECTORY_SEPARATOR . 'autoload.php',
-            \dirname(__DIR__, 3) . \DIRECTORY_SEPARATOR . 'vendor' . \DIRECTORY_SEPARATOR . 'autoload.php',
+            \dirname(__DIR__, 3) . '/vendor/autoload.php',
+            \dirname(__DIR__, 5) . '/autoload.php',
```
Comment (trowski) [inline lib/Threading/Internal/Thread.php]: Is using just `/` compatible with windows?

Comment (kelunik) [inline lib/Threading/Internal/Thread.php]: Yes, I never use `DIRECTORY_SEPARATOR`.

File: lib/Threading/Internal/Thread.php (hunk 1 above)
Comment (kelunik) [inline lib/Threading/Internal/Thread.php]: Done.

File: lib/Threading/Internal/Thread.php (hunk 2 above)
Comment (bwoebi) [inline lib/Threading/Internal/Thread.php]: Why did you switch order? In almost every case (except direct usage), `\dirname(__DIR__, 5) . '/autoload.php'` will be the case matching. Given that there's a `file_exists()` below and we're not doing negative caching in PHP, we'd end up doing a stat(2) call for every run then.

Comment (kelunik) [inline lib/Threading/Internal/Thread.php]: @bwoebi I know, but it might include a file it's not supposed to include.

Comment (bwoebi) [inline lib/Threading/Internal/Thread.php]: @kelunik I then propose another approach: look up in get_included_files() what autoloader has been used and include that. (Do this in constructor, before the new Thread is started)

Comment (kelunik) [inline lib/Threading/Internal/Thread.php]: That can be implemented of top as optimization then.

**Pull request merged**